### ISSUE TITLE
Enhancements for Auto Hyper-Parameter Optimization

### ISSUE THREAD

**Issue by Lemniscate-SHA-256** [Lemniscate-SHA-256/Neural-.nr#52]
Enhancements:
🔹 Add pruning → Stop bad trials early.
🔹 Support more search spaces like dropout rate and weight decay.
🔹 Use distributed tuning to scale across multiple GPUs.

**Comment by Lemniscate-SHA-256** [comment]
Fix for missing dataset issue:
❌ MNIST() is used without being defined.
